sound/soc/sof/intel/hda-dai.c

```diff
@@ -422,56 +422,176 @@ static struct snd_soc_cdai_ops sof_probe_compr_ops = {
 struct snd_soc_dai_driver skl_dai[] = {
 {
 	.name = "SSP0 Pin",
+	.playback = {
+		.stream_name = "ssp0 Tx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
+	.capture = {
+		.stream_name = "ssp0 Rx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
 },
 {
 	.name = "SSP1 Pin",
+	.playback = {
+		.stream_name = "ssp1 Tx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
+	.capture = {
+		.stream_name = "ssp1 Rx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
 },
 {
 	.name = "SSP2 Pin",
+	.playback = {
+		.stream_name = "ssp2 Tx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
+	.capture = {
+		.stream_name = "ssp2 Rx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
 },
 {
 	.name = "SSP3 Pin",
+	.playback = {
+		.stream_name = "ssp3 Tx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
+	.capture = {
+		.stream_name = "ssp3 Rx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
 },
 {
 	.name = "SSP4 Pin",
+	.playback = {
+		.stream_name = "ssp4 Tx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
+	.capture = {
+		.stream_name = "ssp4 Rx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
 },
 {
 	.name = "SSP5 Pin",
+	.playback = {
+		.stream_name = "ssp5 Tx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
+	.capture = {
+		.stream_name = "ssp5 Rx",
+		.channels_min = 2,
+		.channels_max = 2,
+	},
 },
 {
 	.name = "DMIC01 Pin",
+	.playback = {
```

Comment:
you meant capture right for both dmic's?

Reply:
Oh, yes.